File: BSvsZP-Common/Common/StateChange.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Common
{
    public delegate void StateChangeHandler(StateChange change);

    public class StateChange
    {
        public enum ChangeType { NONE, ADDITION, UPDATE, MOVE, DELETION };
        public ChangeType Type { get; set; }
        public object Subject { get; set; }
    }
}

File: BSvsZP-Common -Java/Common/StateChange.java
package Common;

public class StateChange {

    public enum ChangeType {

        NONE, ADDITION, UPDATE, MOVE, DELETION
    };
    public ChangeType Type;
    public Object Subject;

    public ChangeType getType() {
        return Type;
    }

    public void setType(ChangeType type) {
        Type = type;
    }

    public Object getSubject() {
        return Subject;
    }

    public void setSubject(Object subject) {
        Subject = subject;
    }

    public class StateChangeHandler {

        public void addStateChangeHandler() {
            StateChange Changed = new StateChange();
            Changed.Type = StateChange.ChangeType.UPDATE;
            Changed.Subject = this;

        }
    }

}
